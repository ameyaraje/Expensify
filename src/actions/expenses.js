import { v1 as uuidv1 } from 'uuid';

export const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {}) => {
    return {
        type: 'ADD_EXPENSE',
        expense: {
            id: uuidv1(),
            description: description,
            note: note,
            amount: amount,
            createdAt: createdAt
        }
    };
};

export const removeExpense = ({ id } = {}) => {
    return {
        type: 'REMOVE_EXPENSE',
        id: id
    };
};

export const editExpense = (id, updates) => {
    return {
        type: 'EDIT_EXPENSE',
        id: id,
        updates: updates  
    };
};