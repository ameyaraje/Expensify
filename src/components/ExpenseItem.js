import React from 'react';
import { Link } from 'react-router-dom';

import { removeExpense } from '../actions/expenses';

const ExpenseItem = (props) => {
    // console.log(props);
    return (
        <div>
            <p>
                Name: 
                <Link to={`/edit/${props.item.id}`}>
                    {props.item.description}
                </Link>
            </p>
            <p>
                Amount: {props.item.amount}
            </p>
            <p>
                CreatedAt: {props.item.createdAt}
            </p>
        </div>
    );
};

export default ExpenseItem;