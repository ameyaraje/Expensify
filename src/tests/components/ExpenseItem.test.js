import React from 'react';
import { shallow } from 'enzyme';

import expenses from '../fixtures/expenses';
import ExpenseItem from '../../components/ExpenseItem';

test('should render individual expense object', () => {
    const wrapper = shallow(<ExpenseItem item={expenses[0]}/>);

    expect(wrapper).toMatchSnapshot();
});