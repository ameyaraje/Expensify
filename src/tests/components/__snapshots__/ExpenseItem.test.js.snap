// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render individual expense object 1`] = `
<div>
  <p>
    Name:
    <Link
      to="/edit/123"
    >
      bill 1
    </Link>
  </p>
  <p>
    Amount: 
    10
  </p>
  <p>
    CreatedAt: 
    -345600000
  </p>
</div>
`;
